src/main/java/de/retest/recheck/ui/descriptors/RootElementFilter.java

```diff
@@ -0,0 +1,96 @@
+package de.retest.recheck.ui.descriptors;
+
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map.Entry;
+import java.util.stream.Collectors;
+
+import de.retest.recheck.ignore.Filter;
+import de.retest.recheck.ui.image.Screenshot;
+import lombok.RequiredArgsConstructor;
+
+@RequiredArgsConstructor
+public class RootElementFilter {
+	private final Filter filter;
+
+	public RootElement filter( final RootElement rootElement ) {
+		if ( filter.equals( Filter.FILTER_NOTHING ) ) {
+			return rootElement;
+		}
+		final IdentifyingAttributes newIdentifyingAttributes =
+				filter( rootElement, rootElement.getIdentifyingAttributes() );
+		final Attributes newAttributes = filter( rootElement, rootElement.getAttributes() );
+		final String retestId = rootElement.getRetestId();
+		final Screenshot screenshot = rootElement.getScreenshot();
+		final String screen = rootElement.getScreen();
+		final int screenId = rootElement.getScreenId();
+		final String title = rootElement.getTitle();
+		final List<Element> containedElements = rootElement.getContainedElements();
+
+		final RootElement newRootElement = new RootElement( retestId, newIdentifyingAttributes, newAttributes,
+				screenshot, screen, screenId, title );
+
+		final List<Element> newContainedElements = filterContainedElements( newRootElement, containedElements );
+		newRootElement.addChildren( newContainedElements );
+		return newRootElement;
+	}
+
+	public IdentifyingAttributes filter( final Element element, final IdentifyingAttributes identifyingAttributes ) {
+		if ( filter.equals( Filter.FILTER_NOTHING ) ) {
+			return identifyingAttributes;
+		}
+		final List<Attribute> attributes = identifyingAttributes.getAttributes();
+
+		final Collection<Attribute> filteredAttributes = attributes.stream() //
+				.filter( attribute -> !filter.matches( element, attribute.getKey() ) ) //
+				.collect( Collectors.toList() );
+		return new IdentifyingAttributes( filteredAttributes );
+	}
+
+	public Attributes filter( final Element element, final Attributes attributes ) {
+		if ( filter.equals( Filter.FILTER_NOTHING ) ) {
+			return attributes;
+		}
+		final MutableAttributes filteredAttributes = new MutableAttributes();
+		final Iterator<Entry<String, Object>> iterator = attributes.iterator();
+
+		while ( iterator.hasNext() ) {
+			final Entry<String, Object> entry = iterator.next();
+			if ( filter.matches( element, entry.getKey() ) ) {
+				continue;
+			}
+			filteredAttributes.put( entry.getKey(), (String) entry.getValue() );
+		}
+		return filteredAttributes.immutable();
+	}
+
+	public Element filter( final Element element ) {
```

Comment:
Do you really need to filter an single element? Because to be complete, you would need to replace the new child within the parent at the same index once done. Otherwise you would have the new child floating around with only the connection to the parent, but the parent does not now about its child. That would be a sad story 😢.

Reply:
That would be a sad story indeed. But doesn't line 80 ensures that this situation never occurs?

`newElement.addChildren( newContainedElements );`

Reply:
But that would only be the case for the grand children. The `element`s parent does not get updated, because [L67](https://github.com/retest/recheck/blob/9b2503a16ec5ff991fd3223199f61859970f127c/src/main/java/de/retest/recheck/ui/descriptors/RootElementFilter.java#L67) only creates the new child. 

You could argue, that it is the task of the caller to update the method. But on the other hand, is this method really needed at the moment?

Reply:
You are absolutely right. I completely missed that point. As we don't need this method at the moment, I would agree that this needs to go.